### PR TITLE
added pagination, sorting for books. Added OpenApi-ui

### PR DESCRIPTION
**Pull request by IhorKalaur**

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -25,30 +31,38 @@ public class BookController {
...
     @GetMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "get book by id", description = "get book by id if available")
     public BookDto getById(@PathVariable Long id) {
         return bookService.getById(id);
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "create a new book", description = "create a new book")
```
Comment (padu12) [inline src/main/java/com/example/bookstore/controller/BookController.java]: We can use only a summary or description if the text is the same. Please fix it for every method

File: src/main/java/com/example/bookstore/exceptions/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -35,11 +34,11 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
         return new ResponseEntity<>(body, headers, status);
     }
 
-    @ExceptionHandler(Exception.class)
+    /*@ExceptionHandler(Exception.class)
     public ResponseEntity<String> handleAllExceptions(Exception ex) {
         return new ResponseEntity<>("An unexpected error occurred",
                 HttpStatus.INTERNAL_SERVER_ERROR);
-    }
+    }*/
```
Comment (padu12) [inline src/main/java/com/example/bookstore/exceptions/GlobalExceptionHandler.java]: we should avoid pushing commented code

**Review by padu12 — APPROVED** [review]
Amazing✨ You can improve your solution even more)

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**